tests/structured_conf/test_structured_config.py

```diff
@@ -943,7 +953,8 @@ def test_str2str(self, module: Any) -> None:
     def test_dict_subclass_data_preserved_upon_node_creation(self, module: Any) -> None:
         src = module.DictSubclass.Str2StrWithField()
         src["baz"] = "qux"
-        cfg = OmegaConf.structured(src)
+        with self.warns_deprecated():
+            cfg = OmegaConf.structured(src)
```

Comment:
This would be cleaner (you will probably have to move the function to the global scope).
Also, make the name more specific: `warns_dict_subclass_deprecated()`.
```suggestion
        with warns_deprecated():
            cfg = OmegaConf.structured(src)
```

Reply:
Done in https://github.com/omry/omegaconf/pull/707/commits/c6054dbdcdfdcc3e7cdeef9cab6e56c2057574b7.